include/ucl++.h

```diff
@@ -356,46 +424,136 @@ class Ucl final {
 		return out;
 	}
 
-	static Ucl parse (const std::string & in, std::string & err)
+	static Ucl parse (const std::string &in, std::string &err)
 	{
-		auto parser = ucl_parser_new (UCL_PARSER_DEFAULT);
+		return parse (in, std::map<std::string, std::string>(), err);
+	}
 
-		if (!ucl_parser_add_chunk (parser, (const unsigned char *)in.data (),
-				in.size ())) {
-			err.assign (ucl_parser_get_error (parser));
-			ucl_parser_free (parser);
+	static Ucl parse (const std::string &in, const std::map<std::string, std::string> &vars, std::string &err)
```

Comment:
* As far as I read, universal reference is related with type deduction; It seems that  only `T&&` and `auto &&` are called 'universal reference' https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers
* After all, if I change the parameter type to `std::map<std::string, std::string> &&vars`, compiler emits error messages as follows
  - `ucl++.h:477:44: error: cannot bind ‘std::map<std::basic_string<char>, std::basic_string<char> >’ lvalue to ‘std::map<std::basic_string<char>, std::basic_string<char> >&&’`
  - Yes, I passed lvalue `map` instance in my test code.
  - I use g++ (GCC) 4.8.5